mobile/src/main/java/org/openhab/habdroid/ui/OpenHABPreferencesActivity.java

```diff
@@ -224,17 +250,111 @@ public boolean onPreferenceClick(Preference preference) {
-                final PreferenceScreen ps = getPreferenceScreen();
-                ps.removePreference(ps.findPreference(Constants.PREFERENCE_FULLSCREEN));
+            ringtonePreference.setOnPreferenceChangeListener(new OnPreferenceChangeListener() {
+                @Override
+                public boolean onPreferenceChange(Preference preference, Object newValue) {
+                    updateRingtonePreferenceSummary(preference, newValue);
+                    return true;
+                }
+            });
+
+            final PreferenceScreen ps = getPreferenceScreen();
+
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.KITKAT) {
+                Log.d(TAG, "Removing fullscreen pref as device isn't running Kitkat or higher");
+                Preference fullscreenPreference = ps.findPreference(Constants.PREFERENCE_FULLSCREEN);
+                getParent(fullscreenPreference).removePreference(fullscreenPreference);
+            }
+
+            if (!CloudMessagingHelper.isSupported()) {
+                Log.d(TAG, "Removing full-only prefs");
+                getParent(ringtonePreference).removePreference(ringtonePreference);
+                Preference vibrationPreference =
+                        ps.findPreference(Constants.PREFERENCE_NOTIFICATION_VIBRATION);
+                getParent(vibrationPreference).removePreference(vibrationPreference);
+            }
+
+            if (mOpenhabVersion == 1) {
+                Log.d(TAG, "Removing prefs that aren't available in openHAB 1");
+                Preference iconFormatPreference =
+                        ps.findPreference(Constants.PREFERENCE_ICON_FORMAT);
+                getParent(iconFormatPreference).removePreference(iconFormatPreference);
+                Preference chartScalingPreference =
+                        ps.findPreference(Constants.PREFERENCE_CHART_SCALING);
+                getParent(chartScalingPreference).removePreference(chartScalingPreference);
             }
         }
 
+        /**
+         * @author https://stackoverflow.com/a/17633389
+         */
+        private PreferenceGroup getParent(Preference preference)
+        {
+            return getParent(getPreferenceScreen(), preference);
+        }
+
+        /**
+         * @author https://stackoverflow.com/a/17633389
+         */
+        private PreferenceGroup getParent(PreferenceGroup root, Preference preference)
+        {
+            for (int i = 0; i < root.getPreferenceCount(); i++)
+            {
+                Preference p = root.getPreference(i);
+                if (p == preference)
+                    return root;
+                if (PreferenceGroup.class.isInstance(p))
+                {
+                    PreferenceGroup parent = getParent((PreferenceGroup)p, preference);
+                    if (parent != null)
+                        return parent;
+                }
+            }
+            return null;
+        }
```

Comment:
Nit: should fix curly braces in those two methods (put starting ones on old line, add missing ones)

Reply:
Done